### ISSUE TITLE
Display names of each emoji

### ISSUE THREAD

**Issue by madarche** [Sav22999/emoji#21]
Hello. First, thanks for the neat addon!

There's one thing missing for this addon to be even more complete: the display of the names of each emoji.

The need is to be able to select the emoji that best meets the mood one wishes to convey. This need is especially important in the "Smileys" section, in which the user may often hesitate between 2 or 3 emojis which looks very similar.

The solution could be to be able to choose between quite similar emojis by actually displaying their names. That's the solution for example present in *Mattermost*  that displays the name of each emoji in the selection process and also adds an HTML `title` (or *tooltip*) that gets displayed when the user hovers the emojis with their pointer: https://docs.mattermost.com/help/messaging/emoji.html. In the non-libre software sphere, it's also how *Slack* does it.

To sum up, 2 things would be good:
* display names in the selection process
* display names when hovering emojis

Thank you a lot!

**Comment by Sav22999** [comment]
@madarche hi! Thanks for using the addon and thanks to this feedback.

I think that show the name as ":name:" is not useful, beacause (actually) you can write it without the coloum ":" in the search-box.

Anyway, I think the tooltip ("title") is a very useful feature.

What do you think?
And, in case, where should the add-on show to you the "name" in the selection process?

**Comment by madarche** [comment]
@Sav22999 it's your users who are grateful to you 🤗 (emoji from your addon 😉) and thank you for your very fast answer!

You are right, the most interesting point is certainly the tooltip ("title")! Moreover it should be quite straightforward!

Thinking more about it, the name could also be displayed in the "copied" notification, next to the emoji itself. This would give an hint to users that each emoji has a name!

Now about the selection process. It's not as important as the tooltip and the additional idea of the "copied" notification. I was mentioning the selection process of Mattermost to you to show you that emoji names have an interest. So I suggest to forget this point for now!

Thanks again!

**Comment by Sav22999** [comment]
@madarche thanks for explanation.

The name aside the emoji, in the "copied notification" is a bit complicated, just because users can turn on "multi-copy" (just show to you all emojis copied.

Anyway, I'll add the tooltip as soon as possible.

Have a nice Sunday! 😉

**Comment by madarche** [comment]
I didn't know about multi-copy, you just made me discover it!!!

So OK 🤗 Great!

**Comment by Sav22999** [comment]
@madarche I've just published a new update (3.10) which implements also this feature.

Please, tell me it's what you thought so I can close this issue

**Comment by madarche** [comment]
Hello @Sav22999 🙂 The tooltip hints are very neatly displayed, this is great 👍‍‍

But, the tooltip hints are not displayed in the "Most used emojis" section! 🤭

**Comment by Sav22999** [comment]
@madarche yes, it's true.
In the Most used section you can find emojis saved in the browser storage. So, to avoid to use more space I save just the emoji, and also because you shouldn't need to find an emoji if you has it in Most used section 😅, right?

**Comment by madarche** [comment]
@Sav22999 please let me disagree 😟 : I've already used some emojis but I'm not sure they were the right choice, and even sometimes I don't remember their meaning exactly. 

I really do understand how painful this kind of request could be for the developer, here I'm just the voice of a user.

**Comment by Sav22999** [comment]
@madarche sure, no problem. Today I'll try to add this feature. (Unluckily you'll need to add again the emojis to see tooltip, because for now the I didn't save tooltip information.
I hope it will be ok for you... I'll update in some hours, I hope.

Thanks for your important feedback. User feedback are always the most important think, I say this as developer, and you're very polite 😉

**Comment by madarche** [comment]
@Sav22999 there's no hurry at all! And yes I think that having the tooltips displayed in the "Most used emojis" section is worth loosing one's most used emojis once! So it's really OK for me. I hope (and think) it won't upset the other users of your addon too much, when compared to the advantage of having tooltips.

Thanks again for caring for your users 🤗

**Comment by Sav22999** [comment]
@madarche Ok, I've just implemented this feature also in Most used emojis.
I've developed a way to do not lose current Most used emojis, but you will not see any tooltip untill you'll copy again that emoji. So, to get the tooltip of an emoji, you can just copy it (also directly the Most used section), and then you should be able to see its tooltip. Please, tell me it's work or not (the update should arrive in some minutes... usually Mozilla is very fast to update).
You could try, in case, to "Check for updates manually" (in case follow these official steps: https://support.mozilla.org/en-US/kb/how-update-add-ons, so click "Check for Updates).

**Comment by madarche** [comment]
Hello @Sav22999! I've got the updated version of your addon and it works exactly as you've described above. I'm completely satisfied with this evolution 👍‍‍ A big thank you! 🙇‍‍♂️

As far as I'm concerned, this issue can now be closed.

**Comment by Sav22999** [comment]
Perfect
